src/main/java/pl/mantiscrab/budgetr/registration/UserService.java

```diff
@@ -25,15 +26,17 @@ private void throwExceptionIfUserAlreadyExists(UserRegisterDto dto) {
             throw new UserAlreadyExistsException("User with username \"" + dto.username() + "\" already exist");
     }
 
-    private User mapUserDtoToUser(UserRegisterDto dto) {
-        UserRegisterDto dtoWithEncodedPassword = encodePassword(dto);
-        User user = UserMapper.userFromUserRegisterDto(dtoWithEncodedPassword);
+    private User createUser(UserRegisterDto dto) {
+        String encodedPassword = passwordEncoder.encode(dto.password());
+        User user = User.builder()
+                .email(dto.email())
+                .username(dto.username())
+                .password(encodedPassword)
+                .authorities(new HashSet<>())
```

Comment:
better put nothing in here?

Reply:
Did you mean ".authorities(new HashSet<>())" ? I had to, because otherwise it might result in "NullPointerException"